pkg/controller/atlasservice/service.go

```diff
@@ -14,30 +15,32 @@ func (r *AtlasServiceReconciler) discoverServices(ctx *workflow.Context) ([]stat
 	// Try to find the service
 	projects, _, err := ctx.Client.Projects.GetAllProjects(context.Background(), &mongodbatlas.ListOptions{})
 	if err != nil {
-		return nil, workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
+		return nil, workflow.Terminate(workflow.AtlasProjectsListFailed, err.Error())
 	}
 	svcList := []status.AtlasProjectService{}
 	for _, p := range projects.Results {
 
 		clusters, _, err := ctx.Client.Clusters.List(context.Background(), p.ID, &mongodbatlas.ListOptions{})
 		if err != nil {
-			return nil, workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
+			return nil, workflow.Terminate(workflow.AtlasClustersListFailed, err.Error())
 		}
 		clusterList := []status.AtlasClusterService{}
 		for _, cluster := range clusters {
+			connStrTokens := strings.Split(cluster.ConnectionStrings.StandardSrv, "://")
```

Comment:
What happens if we get a cluster back with no StandardSrv? It may be useful to check and make sure the string is in the format we expect (contains `://`) before we split it and call connStrTokens[1] and get a panic b/c the split didn't result in 2 strings.

Reply:
Sounds good. I updated the PR accordingly.

Reply:
Also, I will take care of dbaas operator part as well.